classes/Language.php

```
@@ -1653,16 +1653,24 @@ public static function updateMultilangFromClass($table, $className, $lang)
     private static function updateMultilangFromClassForShop(DataLangCore $classObject, self $lang, Shop $shop)
     {
         $shopDefaultLangId = (int) Configuration::get('PS_LANG_DEFAULT', null, $shop->id_shop_group, $shop->id);
-        $shopDefaultLanguage = new Language($shopDefaultLangId);
+        // During install process the default language may not be available yet
+        if (!empty($shopDefaultLangId)) {
+            $shopDefaultLanguage = new Language($shopDefaultLangId);
+            self::loadAdminTranslatorLocale($shopDefaultLanguage->locale, false);
+        }
 
-        self::loadAdminTranslatorLocale($shopDefaultLanguage->locale, false);
+        // The provided locale should always be loaded though
         self::loadAdminTranslatorLocale($lang->locale, false);
 
         $sfContainer = SymfonyContainer::getInstance();
-        $translator = $sfContainer->get(TranslatorInterface::class);
-        (new EntityTranslatorFactory($translator))
-            ->build($classObject)
-            ->translate($lang->id, $shop->id);
+        if ($sfContainer->has('language')) {
```

Shall we allow this case to be happening in the installation process only, and throw an exception otherwise?
I we don't do this, we may end with a method that nothing because finding the service `language` was failing silently.

---

Actually it's completely wrong I wanted to test if the translator is available not `language` Maybe that's why I have errors in the test (hopefully)

Regarding the exception or the silent fail I agree that an exception would be cleaner, but if we do that it would break the installation process And handling an exception only during installation could be an alternative but I'm not sure we have a way to detect if we are in the installation process 🤔

---

> I'm not sure we have a way to detect if we are in the installation process 🤔

If that helps, from the modules we rely on a constant: `defined('PS_INSTALLATION_IN_PROGRESS')`